### PR TITLE
Restrict proxy access to known hosts only

### PR DESCRIPTION
**Pull request by onursumer**
- Restrict proxy access to known hosts only
- Remove unused endpoints and methods
- Rename dev endpoint and add comments

Need to merge cBioPortal/cbioportal-frontend#4941 and cBioPortal/cbioportal-frontend#4942 before merging this one.

### CODE REVIEW

File: src/main/java/org/cbioportal/proxy/ProxyController.java [hunk 1]
```diff
@@ -226,54 +213,30 @@ private <T> ResponseEntity<T> exchangeData(String body, URI uri, HttpMethod meth
         return restTemplate.exchange(uri, method, new HttpEntity<>(body, httpHeaders), responseType);
     }
 
+    private boolean isKnownHost(String pathInfo) {
+        String[] knownHosts = {
+            this.genomeNexusUrl,
+            this.genomeNexusGrch38Url,
+            "http://mutationaligner.org/api",
```
Comment (inodb) [inline src/main/java/org/cbioportal/proxy/ProxyController.java:220]: do we still use mutationaligner?

File: src/main/java/org/cbioportal/proxy/ProxyController.java [hunk 2]
```diff
@@ -226,54 +213,30 @@ private <T> ResponseEntity<T> exchangeData(String body, URI uri, HttpMethod meth
...
+            "http://docs.cbioportal.org/news"
```
Comment (inodb) [inline src/main/java/org/cbioportal/proxy/ProxyController.java:221]: should we fix cors on the docs site?

File: src/main/java/org/cbioportal/proxy/ProxyController.java (hunk 1 above)
Comment (inodb) [reply src/main/java/org/cbioportal/proxy/ProxyController.java:220]: Looks like it, just to check for existence of this URL:

image

Comment (onursumer) [reply src/main/java/org/cbioportal/proxy/ProxyController.java]: We decided to always show the link. See cBioPortal/cbioportal-frontend#4942

File: src/main/java/org/cbioportal/proxy/ProxyController.java (hunk 2 above)
Comment (onursumer) [reply src/main/java/org/cbioportal/proxy/ProxyController.java]: Looks like Github pages now allows CORS! See cBioPortal/cbioportal-frontend#4941

File: src/main/java/org/cbioportal/proxy/ProxyController.java [hunk 3]
```diff
@@ -49,6 +42,12 @@ public class ProxyController {
     
     @Value("${oncokb.public_api.url:https://public.api.oncokb.org/api/v1}")
     private String oncokbApiUrl;
+
+    @Value("${genomenexus.url.grch38:}")
+    private String genomeNexusGrch38Url;
+
+    @Value("${genomenexus.url:}")
+    private String genomeNexusUrl;
```
Comment (haynescd) [inline src/main/java/org/cbioportal/proxy/ProxyController.java:50]: Where are these fields used?

**Review by haynescd — CHANGES_REQUESTED** [review]

**Review by haynescd — APPROVED** [review]
👍

**Pull request merged**